### ISSUE TITLE
Directory sync error: Last synced notebook and latest remote notebook differ

### ISSUE THREAD

**Issue by apfranzen** [orgzly/orgzly-android#685]
- [x] I have searched for existing issues that may be the same as or related to mine.

I have experienced an issue with using a local repository with sync. I checked [out the source](https://github.com/orgzly/orgzly-android/blob/master/app/src/main/java/com/orgzly/android/sync/BookSyncStatus.kt) to see if that would clear up what may be causing this issue, but I'm still not clear what is causing this issue.
- In plain english, what does this error mean? In what conditions would this error occur?

## Steps to Reproduce
1. I have a directory `org-sync` with several `.org` files
2. add `org-sync` as a local directory sync
3. with a clean install of orgzly on my phone, initiate sync
4. orgzly will import these `.org` files as expected
5. if I edit one of my files in orzly on my phone, then sync, orgzly will throw this error: `Last synced notebook and latest remote notebook differ`

## other facts
1. the repository was linked through the Android Files app and the path is: `content://com.android.providers.downloads.documents/tree/msd%3A73372`
2. the folder and `.org` files were created outsize of orgzly, but are valid .org files (and very simple)
3. this also happened with a `.org` file created in orgzly
4. there is nothing out of the ordinary happening with this folder behind the scenes (sync, etc.)

**Comment by ScottFreeCode** [comment]
Background: "Remote" is the files that aren't Orgzly's internal copy of the notebooks. Somewhat confusingly, that means when your repository is local, the local files on the phone (but outside Orgzly) are referred to as "remote". (This makes more sense when they really are remote, as in the other repository types, which sync with something outside the phone.) This may or may not be at issue here, just throwing it out there in case it helps.

**The error message:** just means that you've editted one of the notebooks both within Orgzly and outside Orgzly since you last synced, or at least _that Orgzly believes so._ There could be cases where no edits are made but something updates the timestamp on the file/notebook; or there could conceivably be a bug (whether in Orgzly or in the phone) where timestamps are seen incorrectly somehow.

If you know yob didn't edit the files outside Orgzly, it may be worth double-checking the modified time on the files in the local folder. If that doesn't reveal that they were updated somehow, I think there are settings in Orgzly to display more of the various timestamps on the notebook list screen. Wouldn't necessarily show what's updating the files (if it is that), but would help confirm or deny _whether_ something is updating them.

**Comment by apfranzen** [comment]
One condition that I found while troubleshooting further is orgzly was not throwing an error, but in the sync directory it was creating a new file on each orgzly sync like below. I'm curious under what condition this might happen and if this lends a hint at what's going on?
```
personal.org
personal.org (1)
personal.org (2)
personal.org (3)
personal.org (4)
personal.org (5)
personal.org (6)
```

**Comment by apfranzen** [comment]
Ok, I believe I found the problem 🎉! I will leave some notes below to hopefully help other folks who run into this in the future. Thanks for taking a look @ScottFreeCode! 

tl:dr: This was caused by a quirk relating to [termux internal and external storage](https://wiki.termux.com/wiki/Internal_and_external_storage). 

To use orgzly (or any other app) correctly with folders/files that you are interacting with using termux, you must access these folders/files using `~/shared/*` in termux. An example would be to access a folder in the `Downloads` folder by this path `~/shared/Download/org-sync` from termux as opposed to `~/downloads/org-sync`. 
Alternatively, when you are selecting a directory in orgzly to sync with, from the Android Files app, you can select Open With => Termux, then select the directory you would like to sync. 

And yes, this seems to work, even using `git` within termux for version control 🎉.

Please let me know if you have any questions, I an happy to help.
